src/ChoiceGroup/ChoiceGroup.tsx

```diff
@@ -0,0 +1,189 @@
+  )
+  const validationChild = React.Children.toArray(children).find(child =>
+    React.isValidElement(child) && child.type === ChoiceGroupValidation ? child : null
+  )
+  const captionChild = React.Children.toArray(children).find(child =>
+    React.isValidElement(child) && child.type === ChoiceGroupCaption ? child : null
+  )
+  const id = useSSRSafeId(idProp)
+  const validationMessageId = validationChild && `${id}-validationMessage`
+  const captionId = captionChild && `${id}-caption`
+  const checkIfOnlyContainsChoiceInputs = () => {
+    const formControlComponentChildren = React.Children.toArray(children)
+      .filter(child => React.isValidElement(child) && child.type === FormControl)
+      .map(formControlComponent =>
+        React.isValidElement(formControlComponent) ? formControlComponent.props.children : []
+      )
+      .flat()
+
+    return React.Children.toArray(formControlComponentChildren)
+      .filter(child =>
+        expectedInputComponents.some(inputComponent => React.isValidElement(child) && child.type === inputComponent)
+      )
+      .every(
+        inputComponent =>
+          React.isValidElement(inputComponent) && (inputComponent.type === Checkbox || inputComponent.type === Radio)
+      )
+  }
+
+  if (!checkIfOnlyContainsChoiceInputs()) {
+    // eslint-disable-next-line no-console
+    console.warn('Only `Checkbox` and `Radio` form controls should be used in a `ChoiceGroup`.')
+  }
+
+  if (!labelChild && !ariaLabelledby) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'A choice group must be labelled using a `ChoiceGroup.Label` child, or by passing `aria-labelledby` to the ChoiceGroup component.'
+    )
+  }
+
+  return (
+    <Slots
+      context={{
+        disabled,
+        required,
+        captionId,
+        validationMessageId
+      }}
+    >
+      {slots => {
+        const isLegendVisible = React.isValidElement(labelChild) && !labelChild.props.visuallyHidden
+
+        return (
+          <ChoiceGroupContext.Provider value={{disabled}}>
+            <div>
+              <Box
+                border="none"
+                margin={0}
+                mb={validationChild ? 2 : undefined}
+                padding={0}
+                {...(labelChild && {
+                  as: 'fieldset',
+                  disabled
+                })}
+                sx={sx}
+              >
+                {labelChild ? (
+                  /*
+                    Placing the caption text and validation text in the <legend> provides a better user
+                    experience for more screenreaders.
+
+                    Reference: https://blog.tenon.io/accessible-validation-of-checkbox-and-radiobutton-groups/
+                  */
+                  <Box as="legend" mb={isLegendVisible ? 2 : undefined} padding={0}>
+                    {slots.Label}
+                    {slots.Caption}
+                    <VisuallyHidden>
```

Comment:
Although it's visually hidden, could this become an issue for both screen readers and crawlers that you're duplicating the validation message markup here? 

For screen-readers specifically, it's probably worth adding an `aria-hidden` here.

Reply:
Also, could this be guarded? So it doesn't generate an empty tag if there's no validation message yet?

Reply:
Good call about making this guarded. Will do 👍